crates/completion/src/completions/unqualified_path.rs

```diff
@@ -79,32 +79,34 @@ fn fuzzy_completion(acc: &mut Completions, ctx: &CompletionContext) -> Option<()
 
     let potential_import_name = ctx.token.to_string();
 
-    let possible_imports =
-        imports_locator::find_similar_imports(&ctx.sema, ctx.krate?, &potential_import_name, 400)
-            .filter_map(|import_candidate| match import_candidate {
-                // when completing outside the use declaration, modules are pretty useless
-                // and tend to bloat the completion suggestions a lot
-                Either::Left(ModuleDef::Module(_)) => None,
-                Either::Left(module_def) => Some((
-                    current_module.find_use_path(ctx.db, module_def)?,
-                    ScopeDef::ModuleDef(module_def),
-                )),
-                Either::Right(macro_def) => Some((
-                    current_module.find_use_path(ctx.db, macro_def)?,
-                    ScopeDef::MacroDef(macro_def),
-                )),
-            })
-            .filter(|(mod_path, _)| mod_path.len() > 1)
-            .filter_map(|(import_path, definition)| {
-                render_resolution_with_import(
-                    RenderContext::new(ctx),
-                    import_path.clone(),
-                    import_scope.clone(),
-                    ctx.config.merge,
-                    &definition,
-                )
-            })
-            .take(20);
+    let possible_imports = imports_locator::find_similar_imports(
+        &ctx.sema,
+        ctx.krate?,
+        &potential_import_name,
+        50,
```

Comment:
I wonder what heuristic is used by IntelliJ here to limit the set of auto-import candidates?

Reply:
I'll look at it at some point.

Reply:
Looks like the results are not limited there at all.

This is the analogue of our `imports_locator`: https://github.com/intellij-rust/intellij-rust/blob/48d051f0b040e4cad5fc4dac7f07578e801c808e/src/main/kotlin/org/rust/ide/utils/import/ImportCandidatesCollector.kt

and its usage related to completions seems to be here:
https://github.com/intellij-rust/intellij-rust/blob/48d051f0b040e4cad5fc4dac7f07578e801c808e/src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt#L198

I think we can safely remove the limit from the `imports_locator::find_similar_imports`  invocation, but still have to have the `.take(40)` or similar in the `unqualified_path::fuzzy_completion` function to avoid excessive insert imports calculation.